src/main/java/startsteps/ECommerceShop/config/SecurityConfig.java

```diff
@@ -47,7 +47,8 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                                 "products/name/{name}",
                                 "/cart/add",
                                 "/cart/mycart",
-                                "/cart/remove/{productId}").permitAll()
+                                "/cart/remove/{productId}",
+                                "/order/place").permitAll()
```

Comment:
I think this should not be permitAll. A user should be authenticated if he want to place an order right?

Reply:
yes,  in my controller I have @PreAuthorize("hasAuthority('USER')"), that allows only user place an order.